packages/hg/src/node/hg-impl.ts

```diff
@@ -679,12 +661,12 @@ export class HgImpl implements Hg {
  */
 const logTemplate: string =
     '\\{ "node": "{node}",\\n' +
-    '  "author": "{author}",\\n' +
+    '  "author": {json(author)},\\n' +
     '  "timestamp": "{date|hgdate}",\\n' +
-    '  "added": "{file_adds}",\\n' +
-    '  "modified": "{file_mods}",\\n' +
-    '  "deleted": "{file_dels}"\n}' +
-    'end-json-start-desc{desc}';
+    '  "added": {json(file_adds)},\\n' +
+    '  "modified": {json(file_mods)},\\n' +
+    '  "deleted": {json(file_dels)},\\n' +
+    '  "desc": {json(desc)}\n}';
```

Comment:
A question about this format. I've seen that @mcgordonite wrote that he created a different template [here](https://github.com/ARMmbed/theia/commit/e2d9918c0d0f4b2ee1188d911bdc2b97d9985f5e). this looks slightly different. Is it by design?

Reply:
I think either are fine: this version has the advantage of being trivial to parse with `JSON.parse`, but the one in your link didn't require as much escaping in the template.